analyzeme/src/profiling_data.rs

```diff
@@ -78,33 +81,17 @@ impl ProfilingData {
         assert!(event_byte_count % RAW_EVENT_SIZE == 0);
         event_byte_count / RAW_EVENT_SIZE
     }
-}
-
-pub struct ProfilerEventIterator<'a> {
-    data: &'a ProfilingData,
-    forward_event_idx: usize,
-    backward_event_idx: usize,
-}
-
-impl<'a> ProfilerEventIterator<'a> {
-    pub fn new(data: &'a ProfilingData) -> ProfilerEventIterator<'a> {
-        ProfilerEventIterator {
-            data,
-            forward_event_idx: 0,
-            backward_event_idx: data.num_events(),
-        }
-    }
 
-    fn decode_event(&self, event_index: usize) -> Event<'a> {
+    pub fn decode_full_event<'a>(&'a self, event_index: usize) -> Event<'a> {
```

Comment:
I'd like to keep this `pub(crate)` for now as I don't want to guarantee fast random access to events in the profiling data. That would make it harder for us to ever switch to a variable length encoding or having a single file for events + stringtable.

Reply:
Hi @michaelwoerister , for keeping this function private, I tries to add a function pointer `event_decoder` to LightweightEvent.

I could not confirm if it is the right solution. Please help review the update code. Thanks a lot.